### PR TITLE
attach token, schemaName when calling createForm. add comments and refac...

### PR DESCRIPTION
**Pull request by owen-d**
...tor.

### CODE REVIEW

File: server/schemas/schemaModel.js [hunk 1]
```diff
@@ -45,9 +45,11 @@ module.exports = {
       });
   },
 
-  //findSchema accepts a query object, a stream, and a storage object.
-  //It then attempts to find a matching schema using the query,
-  //stores the template on the storage (if found), and emits the proper event on the stream.
+  /*
```
Comment (dduugg) [inline server/schemas/schemaModel.js]: use [jsdoc](http://usejsdoc.org/) comments

File: server/schemas/schemaModel.js [hunk 2]
```diff
@@ -13,31 +13,31 @@ bluebird.promisifyAll(Model);
 
 module.exports = {
   model: Model,
+  /*
```
Comment (dduugg) [inline server/schemas/schemaModel.js]: use [jsdoc](http://usejsdoc.org/) comments

File: server/schemas/schemaModel.js [hunk 3]
```diff
@@ -13,31 +13,31 @@ bluebird.promisifyAll(Model);
...
-            name: newSchema.name,
-            metaData: newSchema.metaData,
-            data: newSchema.data
-          })
+        if (!schema) {
+          return Model.createAsync(params)
           .then(function(schema) {
             next(false, schema);
           });
+        } else {
```
Comment (dduugg) [inline server/schemas/schemaModel.js]: you don't need an `else` statement if the `if` statement returns, correct?

File: server/schemas/schemaModel.js [hunk 4]
```diff
@@ -13,31 +13,31 @@ bluebird.promisifyAll(Model);
...
     var error;
-    var newSchema = request.body.schema;
-    if (!newSchema.name) {
-      error = {message: "Invalid schema name!"};
-      next(error, false);
-      return;
-    }
-    Model.findOneAsync({name: newSchema.name})
+    var newSchema = request.body;
+    var params = {
```
Comment (dduugg) [inline server/schemas/schemaModel.js]: it seems like `params` is actually the new schema, and `newSchema` is just `body` or `requestBody` or something similar. but i might not understand what's going on...

File: server/schemas/schemaModel.js (hunk 3 above)
Comment (owen-d) [inline server/schemas/schemaModel.js]: If there is not a match found by mongoose, it does not error, but just returns a falsy value for schema (null if I recall correctly). Thats why I explicitly handle this case.

File: server/schemas/schemaModel.js (hunk 4 above)
Comment (owen-d) [inline server/schemas/schemaModel.js]: correct. Do you feel the lable confusing?

File: server/schemas/schemaModel.js (hunk 3 above)
Comment (dduugg) [inline server/schemas/schemaModel.js]: I meant that I think you can simplify the code from:

```
} else {
    next({message: "Schema with that name already exists"}, false);
}
```
to:
```
}
next({message: "Schema with that name already exists"}, false);
```

File: server/schemas/schemaModel.js (hunk 4 above)
Comment (dduugg) [inline server/schemas/schemaModel.js]: I think the suggested `var` renames are clearer but I defer to your judgment.

File: server/schemas/schemaModel.js [hunk 5]
```diff
@@ -14,40 +14,46 @@ bluebird.promisifyAll(Model);
...
 
+  /**
+   *createTemplate:
+   * @param {Object} request the http ClientRequest object
+   * @param {Object} response the http ServerResponse object
+   * @param {Function} next callback to be executed with error and schema arguments.
+   * Accepts incoming request, response, and next (callback).
+   * Attempts to find an existing schema with the provided name. If one does not exists,
+   * a schema template is created from the request parameters.
+  */
```
Comment (dduugg) [inline server/schemas/schemaModel.js]: i don't think this is valid. did you run jsdoc on the project and look at the result?

**Comment by dduugg** [comment]
jsdoc still doesn't work for me on schemaModel, but i'm ok with a merge once everyone signs off

**Comment by owen-d** [comment]
Ok, I've updated the comment code. jsDoc should work now for all but the final function in schemaModel.

**Comment by cmperez** [comment]
:+1:

**Comment by zindlerb** [comment]
:+1:

**Comment by rachelsison** [comment]
:+1:

**Pull request merged**